code/modules/projectiles/guns/mods/mods.dm

```diff
@@ -406,7 +406,39 @@
 	I.gun_loc_tag = GUN_SCOPE
 	I.removable = FALSE
 
+//broken gun mods, for firearms beyond repair
+
+/obj/item/gun_upgrade/barrel/broken
+	name = "Plugged barrel"
+	desc = "A good way to explode your gun, if you could actually chamber anything."
+	icon_state = "Forged_barrel"
+	spawn_blacklisted = TRUE
+	price_tag = 0
+
+/obj/item/gun_upgrade/barrel/broken/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.weapon_upgrades = list(GUN_UPGRADE_BROKEN = TRUE)
+	I.destroy_on_removal = TRUE
+	I.gun_loc_tag = GUN_BARREL
+	I.removable = FALSE
+
+/obj/item/gun_upgrade/mechanism/broken
+	name = "Broken fire control group"
```

Comment:
```suggestion
	name = "broken fire control group"
```
what is even fire control group? receiver maybe?

Reply:
is now broken bolt